src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -21,6 +21,7 @@ public abstract class CollectionTableValue<T> : TableValue
         // Additional capabilities. 
         private readonly IReadOnlyList<T> _sourceIndex; // maybe null. supports index. 
         private readonly IReadOnlyCollection<T> _sourceCount; // maybe null. supports count;
+        private readonly List<T> _sourceList;
```

Comment:
>List [](http://example.com/codeflow?start=25&length=4)

IList<T>

Reply:
The challenge is that we want Append (), and IList doesn't have that, only List. 
We may need to find yet another interface.

Reply:
IList has Add(), which is what it looks like you're using.

Reply:
Changed to ICollection<T>.